docs/developer-handbook.md

```diff
@@ -265,6 +296,319 @@ considered a chore that we are doing to keep things up-to-date.
 
 </details>
 
+## Coding style
```

Comment:
Should we move this under the components part above?

Reply:
We can, it's really up to whatever everybody wants. It is a pretty large section so it might be nice if it is its own thing

Reply:
Does it only apply to carbon-components?